pygef/grouping.py

```diff
@@ -1,46 +1,45 @@
+import pandas as pd
 import polars as pl
 
 
 class GroupClassification:
     def __init__(self, zid, df, min_thickness):
+        df_group = df.to_pandas().copy()
         self.zid = zid
-        start_depth = df["depth"][0]
-        df_group = df[:, ["depth", "soil_type"]]
-
-        self.df_group = (
-            df_group.pipe(self.group_equal_layers, "soil_type", "depth", start_depth)
-            .pipe(group_significant_layers, min_thickness, start_depth)
-            .pipe(self.group_equal_layers, "layer", "zf", start_depth)
+        start_depth = df_group["depth"][0]
+        df_group = df_group.loc[:, ["depth", "soil_type"]]
```

Comment:
Shouldn't this be the following?

```suggestion
        df_group = pl.from_pandas(df_group.loc[:, ["depth", "soil_type"]])
```

Reply:
it's already a pandas DataFrame

Reply:
I know, but in the next line you create a polars dataframe from it:

```python
self.df_group = pl.DataFrame(..
```